pages/api/[entity_type_slug]/index.js

```diff
@@ -0,0 +1,52 @@
+import { withSession } from "@klaudsol/commons/lib/Session";
+import RecordNotFound from "@klaudsol/commons/errors/RecordNotFound";
+import { OK, NOT_FOUND } from "@klaudsol/commons/lib/HttpStatuses";
+import { setCORSHeaders, handleRequests } from "@klaudsol/commons/lib/API";
+import { assertUserCan } from '@klaudsol/commons/lib/Permissions';
+import { createHash } from "@/lib/Hash";
+import { readContents } from '@/lib/Constants';
+import Entity from "@/backend/models/dynamodb/Entity";
+import { 
+    formatAttributesData, 
+    formatContentData, 
+    getAttributeMap, 
+    getEntityVariant 
+} from "@/utils/dynamodb/formatResponse";
+import { DYNAMO_DB_ID_SEPARATOR } from "@/constants";
+
+export default withSession(handleRequests({ get }));
+
+async function get(req, res) {
+    await assertUserCan(readContents, req);
+
+    const { entity_type_slug } = req.query;
+
+    const rawData = await Entity.whereAll({ entity_type_slug });
+    const rawMetadata = await Entity.whereMetadata({ entity_type_slug });
```

Comment:
I think `metadata` may be a field with type Map. In our case, both Entities and Entity Types have metadata anyway.

This have an advantage of being able to access the metadata as `rawData.Items[0].metadata.M`

Reply:
Replaced this with `Entity.whereEntityType` method. This returns all of the fields where `type = 'content'` and `slug = :entity_type_slug`. 

I have also replaced the `metadata` field with `attributes` field.